### ISSUE TITLE
Range of CosineSimilarity[-1,1] dose not match the label(0,1)

### ISSUE THREAD

**Issue by weiyu322** [UKPLab/sentence-transformers#25]
I find in the code, the textual semantic similarity tasks like sts,  CosineSimilarityLoss first calculates the cosine distance between two vectors , and then uses mean squared error  to calculate the loss.
However the range of cosine distance is [-1,1](<0:not similar, >0:similar) and the labels are in (0,1), it seems that at training step, the model tries to map space:[-1,1] to space[0,1]. And at the prediction step, >0.5 means similar, <0.5 not similar.
i do not know if i understand it right

**Comment by nreimers** [comment]
Hi @weiyu322 
I experimented with different versions, for example, with scaling scores to -1...1, or to use sigmoid instead of cosine to get scores between 0...1. But cosine similarity, MSE and the original scores 0...1 worked best (gave the best scores on the STS benchmark dataset).

Ensuring that cosine similarity scores gives results between 0...1 is rather easy, for example, you get this, if all your dimensions are positive.

You should not conclude that <0 => not similar and >0 => similar (or <0.5 not similar and >0.5 similar). What is considered similar and what is not considered dissimilar depends on your task. Further, it is better not to assume that the scale is linear.

For example, you might get a similarity metric that produces only scores between 0 and 0.1, or maybe between -0.3 and 0.7. You never know how your similarity function behaves.

It is better just to assume that a higher score means more similar and a lower score means less similar. Which parts of the scale means now similar and which means dissimilar depends on your task and the function.
